### PR TITLE
Net6+VS2022 support 

### PR DESCRIPTION
**Pull request by jasonswearingen**
# PR Details

Net6+VS2022 support.   Should be **put into a separate branch** for now.  The following things need to be fixed before putting in master:
 - [ ] VS2022 support only.  VS2019 is broken due to https://github.com/stride3d/stride/issues/1119  should support both
 - [ ] VSPlugin build is broken.  Its build process needs to be updated to a x64 system.
 - [ ] Net6 is in preview.  there may be breaking changes to the runtime in the next couple months before it's RTW

## Description

upgrade stride to target DotNet6.0

**IT IS DONE/COMPLETE**.   I'm done so it's really just up to Stride contribs to decide when/where to merge.  

## Related Issue

see https://github.com/stride3d/stride/issues/1125 for detailed description

## Motivation and Context

please see the related issue

## Types of changes

- [X] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [X] My change requires a change to the documentation.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: sources/core/Stride.Core.Design/Stride.Core.Design.csproj [hunk 1]
```diff
@@ -27,8 +27,9 @@
   <ItemGroup>
     <PackageReference Include="Microsoft.CSharp" Version="4.7.0" />
     <PackageReference Include="Microsoft.VisualStudio.Setup.Configuration.Interop" Version="2.3.2262-g94fae01e" />
-    <PackageReference Include="Microsoft.Extensions.DependencyModel" Version="2.1.0" />
-    <PackageReference Include="NuGet.Configuration" Version="5.8.0" />
+    <PackageReference Include="Microsoft.Extensions.DependencyModel" Version="6.0.0-preview.7.21377.19" />
+    <PackageReference Include="Newtonsoft.Json" Version="12.0.3" />
```
Comment (Eideren) [inline sources/core/Stride.Core.Design/Stride.Core.Design.csproj:31]: This change is suspicious, ``Core.Design`` needs ``Newtonsoft.Json`` and it was previously packaged with the two other references ?

File: sources/editor/Stride.Editor/Stride.Editor.csproj [hunk 2]
```diff
@@ -26,6 +26,7 @@
   </ItemGroup>
   <ItemGroup>
     <ProjectReference Include="..\..\assets\Stride.Core.Assets.CompilerApp\Stride.Core.Assets.CompilerApp.csproj" />
+    <ProjectReference Include="..\..\engine\Stride.Games\Stride.Games.csproj" />
```
Comment (Eideren) [inline sources/editor/Stride.Editor/Stride.Editor.csproj:29]: Kind of suspicious here as well, did you commit this by mistake ?

**Review by Eideren — COMMENTED** [review]
Thanks ! 
Did you try to create a project under net5 and then open that project with the net6 gamestudio ?
You mentioned the vs plugin being broken over on #1125 is that building the plugin or using it inside of VS. 
If it's building then I would think we should take care of that before we merge this in.

File: sources/core/Stride.Core.Design/Stride.Core.Design.csproj (hunk 1 above)
Comment (jasonswearingen) [reply sources/core/Stride.Core.Design/Stride.Core.Design.csproj:31]: yes, ```Newtonsoft.Json``` was bundled inside of ```"Microsoft.Extensions.DependencyModel" Version="2.1.0"```

File: build/Stride.sln.DotSettings [hunk 3]
```diff
@@ -27,6 +27,7 @@ Distributed under the MIT license. See the LICENSE.md file in the project root f
 	<s:Boolean x:Key="/Default/Environment/SettingsMigration/IsMigratorApplied/=JetBrains_002EReSharper_002EPsi_002ECSharp_002ECodeStyle_002ECSharpKeepExistingMigration/@EntryIndexedValue">True</s:Boolean>
 	<s:Boolean x:Key="/Default/Environment/SettingsMigration/IsMigratorApplied/=JetBrains_002EReSharper_002EPsi_002ECSharp_002ECodeStyle_002ECSharpPlaceEmbeddedOnSameLineMigration/@EntryIndexedValue">True</s:Boolean>
 	<s:Boolean x:Key="/Default/Environment/SettingsMigration/IsMigratorApplied/=JetBrains_002EReSharper_002EPsi_002ECSharp_002ECodeStyle_002ECSharpRenamePlacementToArrangementMigration/@EntryIndexedValue">True</s:Boolean>
+	<s:Boolean x:Key="/Default/Environment/SettingsMigration/IsMigratorApplied/=JetBrains_002EReSharper_002EPsi_002ECSharp_002ECodeStyle_002ECSharpUseContinuousIndentInsideBracesMigration/@EntryIndexedValue">True</s:Boolean>
```
[the remark this answers is not in the document]
Comment (jasonswearingen) [reply build/Stride.sln.DotSettings:30]: This is auto added by Resharper for VS2022 with default settings.   I agree it's not needed, but shouldn't all this Resharper stuff be removed?

File: sources/editor/Stride.Editor/Stride.Editor.csproj (hunk 2 above)
Comment (jasonswearingen) [reply sources/editor/Stride.Editor/Stride.Editor.csproj]: right, not sure how that ref got added.  removing it and things still build

**Comment by jasonswearingen** [comment]
> Did you try to create a project under net5 and then open that project with the net6 gamestudio ?
Yes that works fine.   However I just discovered that running a project from within GameStudio fails due to nuget not finding a Microsoft.Net.HostModel assembly v6.0.0.   It seems to be a sub-reference and I can't seem to solve this.

> You mentioned the vs plugin being broken over on #1125 is that building the plugin or using it inside of VS.
> If it's building then I would think we should take care of that before we merge this in.
It is not building.  building fails with 
```
Severity	Code	Description	Project	File	Line	Suppression State
Error	VSSDK1000	Failed to load 'C:\Users\Jason\.nuget\packages\microsoft.vssdk.buildtools\15.0.26201\tools\VSSDK\bin\VSCT.exe' Assembly. Could not load file or assembly 'file:///C:\Users\Jason\.nuget\packages\microsoft.vssdk.buildtools\15.0.26201\tools\VSSDK\bin\VSCT.exe' or one of its dependencies. An attempt was made to load a program with an incorrect format.	Stride.VisualStudio.Package	C:\Users\Jason\.nuget\packages\microsoft.vssdk.buildtools\15.0.26201\tools\VSSDK\Microsoft.VsSDK.Common.targets	128	
```

**Comment by jasonswearingen** [comment]
I don't have the expertise to fix the VSPlugin or the Gamestudio running project problems.

File: build/Stride.sln.DotSettings (hunk 3 above)
[the remark this answers is not in the document]
Comment (Eideren) [reply build/Stride.sln.DotSettings:30]: There's a config in there that auto-includes the copyright header when creating new files, haven't looked into the rest but at least that one is necessary.

**Comment by jasonswearingen** [comment]
@xen2 could you please make a ```vs2022-net6``` branch for this to be pulled into.  There are some outstanding issues beyond my ability to fix, plus some ecosystem incompatibilities that should resolve by the time Net6 goes to RTW

**Comment by jasonswearingen** [comment]
**update**: Running game studio projects works if done through the Game Studio stand-alone (run from the launcher, not via Visual Studio).  So the only thing that needs fixing is the VS Plugin.

**Comment by xen2** [comment]
Nice work!
I will integrate it directly as a branch (that we can merge in master as soon as .NET 6 is officially released)

**Comment by xen2** [comment]
Longer term we might also want to switch mobile to net6.0-android/ios (of course, not part of this PR and we can still release with current TFM for mobile so nothing mandatory)

**Comment by jasonswearingen** [comment]
This PR is the minimum to get net6+VS2022 working.

additional unnecessary updates can be found in https://github.com/jasonswearingen/stride/tree/net6-unnecessary-build-updates

The maintainers may consider merging those also.

**Comment by jasonswearingen** [comment]
merged xen's fix for UpdateEngine broken (see #1136 )
also applied other bugfixes to CompilerApp changes made in master branch.   now working again.

**Comment by jasonswearingen** [comment]
merged stride/master work.   auto-merge.  didn't test, but glance at the changes show nothing related.

**Comment by xen2** [comment]
Regarding VS2019 support:
#1119 is supposedly fixed in preview 5+ and RC3+
https://developercommunity.visualstudio.com/t/C-projects-are-targeting-Visual-Studio/1506031#T-N1565178

**Comment by xen2** [comment]
Gave a quick try to latest VS2022 + .NET 6
Ran into some issues with NuGet dll clashes between ours and MSBuild (again!).

I think one way to fix that once and for all would be to call MSBuild using out-of-process nodes.
However it looks like it's currently broken with .NET Core. I investigated and submitted a PR:
https://github.com/dotnet/msbuild/pull/7013
Crossing fingers that it can be merged in time to be in .NET Core 6.0.1xx dailies soon and VS2022 RTM.

**Pull request merged**